stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -1795,6 +1805,152 @@ impl NakamotoChainState {
         }
     }
 
+    pub fn calculate_signer_slots(
+        clarity: &mut ClarityTransactionConnection,
+        pox_constants: &PoxConstants,
+    ) -> Result<HashMap<Value, u128>, clarity::vm::errors::Error> {
+        let is_mainnet = clarity.is_mainnet();
+        let pox4_contract = &boot_code_id(POX_4_NAME, is_mainnet);
+        let reward_cycle = clarity
+            .eval_read_only(pox4_contract, &"(current-pox-reward-cycle)")
```

Comment:
I'd prefer if you just passed `reward_cycle` as an argument.  The code that calls this -- the block-processing logic -- ought to be able to calculate the reward cycle from `pox_constants`.